### PR TITLE
Update to 20.10.6

### PR DESCRIPTION
**Pull request by tianon**
This is an alternative/upgrade to #13 to see how bad the damage is. :weary: :speak_no_evil:

### CODE REVIEW

**Comment by tianon** [comment]
This error message is hilarious to read, knowing it's being read from `journald`:

> `2021-02-03T19:43:22Z docker.dockerd[12199]: time="2021-02-03T19:43:22.960766724Z" level=error msg="Handler for POST /v1.41/containers/943cf392d62f2c5ebcfd32909488f9265fd2a60bbbe9268e64bfce5add4af822/start returned error: failed to initialize logging driver: journald is not enabled on this host"`

It comes from https://github.com/moby/moby/blob/v20.10.3/daemon/logger/journald/journald.go#L60, but the only substantive change to that file was updating from `github.com/coreos/go-systemd/journal` to `github.com/coreos/go-systemd/v22/journal`; looking at the diff in https://github.com/coreos/go-systemd/compare/39ca1b05acc7ad1220e09f133283b8859a8b71ab...2d78030078ef61b3cae27f42ad6d0e46db51b339#diff-724313773be2f14993c9b6e81803e33eb071b561d075f59bee32a5b2a09fa8f7, it seems the only real functional change there is `net.Dial` converted to `net.ListenUnixgram`. :confused:

**Comment by tianon** [comment]
Ooof, `apparmor="DENIED" operation="ptrace" profile="snap.docker.dockerd" pid=12948 comm="ps" requested_mask="trace" denied_mask="trace"` over and over and over and over (many, many times per second), but after filtering out all that noise, this looks relevant:

```
2021-02-03T20:18:38.0225521Z Feb 03 20:18:34 fv-az175-209 audit[12896]: AVC apparmor="DENIED" operation="bind" profile="snap.docker.dockerd" pid=12896 comm="exe" family="unix" sock_type="dgram" protocol=0 requested_mask="bind" denied_mask="bind" addr=none
2021-02-03T20:18:38.0227497Z Feb 03 20:18:34 fv-az175-209 kernel: audit: type=1400 audit(1612383514.377:425): apparmor="DENIED" operation="bind" profile="snap.docker.dockerd" pid=12896 comm="exe" family="unix" sock_type="dgram" protocol=0 requested_mask="bind" denied_mask="bind" addr=none
2021-02-03T20:18:38.0229429Z Feb 03 20:18:34 fv-az175-209 audit[12903]: AVC apparmor="DENIED" operation="bind" profile="snap.docker.dockerd" pid=12903 comm="exe" family="unix" sock_type="dgram" protocol=0 requested_mask="bind" denied_mask="bind" addr=none
2021-02-03T20:18:38.0231357Z Feb 03 20:18:34 fv-az175-209 kernel: audit: type=1400 audit(1612383514.645:426): apparmor="DENIED" operation="bind" profile="snap.docker.dockerd" pid=12903 comm="exe" family="unix" sock_type="dgram" protocol=0 requested_mask="bind" denied_mask="bind" addr=none
2021-02-03T20:18:38.0233250Z Feb 03 20:18:34 fv-az175-209 audit[12910]: AVC apparmor="DENIED" operation="bind" profile="snap.docker.dockerd" pid=12910 comm="exe" family="unix" sock_type="dgram" protocol=0 requested_mask="bind" denied_mask="bind" addr=none
2021-02-03T20:18:38.0235199Z Feb 03 20:18:34 fv-az175-209 kernel: audit: type=1400 audit(1612383514.829:427): apparmor="DENIED" operation="bind" profile="snap.docker.dockerd" pid=12910 comm="exe" family="unix" sock_type="dgram" protocol=0 requested_mask="bind" denied_mask="bind" addr=none
```

I bet I need to connect the interface before I try to use the `journald` logger, and possibly even restart the Docker daemon, but it *is* strange that it can't even detect whether or not `journald` is enabled anymore. :disappointed:

Looks like https://github.com/coreos/go-systemd/issues/331 is likely related and/or discussion of this very issue.

**Comment by anonymouse64** [comment]
Regarding the go-systemd issue, I have filed another update to docker-support, just adding the same rule that we already provide to kubernetes-support for the same reason: https://github.com/snapcore/snapd/pull/9924, I hope to get that one into 2.50

**Comment by tianon** [comment]
Ugh, https://github.com/moby/buildkit/pull/1533 breaks Git via BuildKit again...

(specifically https://github.com/moby/buildkit/pull/1533/files#diff-9a3248027714dcbcf2647fa27b75ec4dd3d9639ae80c40bfd6a68abd7805bd4eR494-R498 clears `git`'s environment and didn't previously, so we'll have to change how we make sure `git` gets our `GIT_*` variables now)

**Comment by tianon** [comment]
(Now that I've got #27 working, I've rebased this on that to see if that helps here.)

**Comment by tianon** [comment]
Ugh, we get bit by the same change because we lose Snappy's `LD_LIBRARY_PATH`.

**Comment by tianon** [comment]
(Was hoping I could solve this without a patch, but as seen by the recent force push history here, no dice hahaha)

File: .github/workflows/smoke-test.yml [hunk 1]
```diff
@@ -31,6 +31,10 @@ jobs:
           sudo snap refresh
           sudo snap list
 
+          # get the latest beta snapd (to satisfy our "assumes:")
+          sudo snap refresh --beta snapd
+          # TODO remove "--beta" once 2.50 hits stable
```
Comment (tianon) [inline .github/workflows/smoke-test.yml:36]: Not sure what to do about this -- maybe we wait to merge this until 2.50 is stable?

Comment (tianon) [reply .github/workflows/smoke-test.yml:36]: ... oh, 2.50 hit candidate yesterday, so it's probably a short time before it's stable anyhow :smile: :innocent:

Comment (anonymouse64) [reply .github/workflows/smoke-test.yml:36]: plan is for 2.50 to hit stable on may 10th, we have a company-wide sprint next week, hence the week after

**Comment by tianon** [comment]
Looks like we might finally be good??

**Comment by anonymouse64** [comment]
2.50 is now in stable BTW

**Comment by tianon** [comment]
Appreciate the review! :heart:

**Pull request merged**